usecase/update_todo_usecase.go

```diff
@@ -0,0 +1,40 @@
+package usecase
+
+import (
+	"context"
+	"time"
+	"todo-clean/common"
+	"todo-clean/domain"
+	"todo-clean/lib/errorLib"
+)
+
+func (n *newUseCase) UpdateTodoUseCase(ctx context.Context, queryEntity domain.QueryUpdateTodoEntity, entity domain.UpdateTodoEntity) (result *domain.UpdateTodoEntity, err error) {
+
+	response, err := n.repo.GetTodoRepository(ctx, &queryEntity.ID)
+	if len(response) == 0 {
+		return nil, errorLib.WrapError(common.ErrDataNotFound.Error(), err)
+	}
+
+	dbTx, err := n.repo.Begin()
+	if err != nil {
+		return nil, errorLib.WrapError(common.ErrBeginTodo.Error(), err)
+	}
+	result, err = n.repo.UpdateTodoRepository(ctx, dbTx, queryEntity, domain.UpdateTodoEntity{
+		Title:       entity.Title,
+		Description: entity.Description,
+		Status:      entity.Status,
+		UpdatedAt:   time.Now(),
+	})
+	if err != nil {
+		return nil, errorLib.WrapError(common.ErrUseCaseUpdateTodo.Error(), err)
```

Comment:
If error you must rollback transaction

Reply:
i use rollback in line 33

Reply:
In line 33 will handle from error repo.Commit
That's why you need to rollback in repo.UpdateToDoRepository too